File: converter.py
import sys
import json
import os
import re

args = sys.argv;

current_file_name = args[1] if len(args) > 1 else 'csv.csv'
new_file_name = args[2] if len(args) > 2 else 'json.json'

try:
	file = open(current_file_name, 'r').read()
except: 
	print "File was not found."
	exit()

file = file.split('\n')


for i in range(len(file)):
	if file[i] == '':
		file.remove('');


headers_string = file[0]

headers_array = headers_string.split(',')

rows_array = file[1:]

bad = 0

master = []

for i in range(len(rows_array)):
	row = re.sub(r',((?!")(?! ))', '," "', rows_array[i])

	row = re.sub(r'(?<="),', '^^^', row)

	split_row = row.split('^^^')

	data = {}

	for j in range(len(split_row)):
		data[headers_array[j]] = split_row[j]

	master.append(data)


new_file = open(new_file_name, 'w')

try:
	new_file.write(json.dumps(master))
except:
	os.remove(new_file_name)
	print 'Something went wrong. The file was not saved.'
	exit()


print 'JSON was successfully saved to %s' %(new_file_name)

File: converter.js
const fs = require('fs');

const args = process.argv;

var file_name = args[2] ? args[2] : './csv.csv';

var new_file_name = args[3] ? args[3] : './json.json';

// Get file
fs.readFile(file_name, 'utf8', (error, res) => {
	if(error) {
		console.log(error.code == 'ENOENT' ? 'File was not found.' : 'Something went wrong, please try again.');
		return;
	}

	// Parse headers
	let headers_string = res.split(/\n/)[0];
	let headers_array = headers_string.split(',');


	// Parse rows
	let data = res.split(/\n/);

	data.shift();

	let data_array = data;

	// Remove empty string items.
	// Some csv files give two line breaks instead of one, leaving an empty string in between actual data
	for(i = 0; i < data_array.length; i++) {
		if(data_array[i] === '') {
			data_array.splice(i, 1);
			i--;
		}
	}


	// Join corresponding headers and rows into an object
	let rows = [];
	for(i = 0; i < data_array.length; i++) {
		// Fix more than one sequential comma
		data_array[i] = data_array[i].replace(/,((?!")(?! ))/g, '," "');

		data_array[i] = data_array[i].replace(/(?<="),/g, '^^^');

		data_array[i] = data_array[i].split('^^^');

		let object = {};

		for(j = 0; j < data_array[i].length; j++) {
			object[headers_array[j]] = data_array[i][j];
		}
		console.log(data_array[i].length, i);

		rows.push(object);
	}

	// Write JSON to new file
	fs.writeFile(new_file_name, JSON.stringify(rows), 'utf8', (error) => {
		if(error) {
			console.log('Something went wrong when saving the JSON file.');
			return;
		}

		console.log(`The JSON file was saved as ${new_file_name}`);
	});
});
